### DISCUSSION TITLE
Wrk/Wrk2 support extra pipelining option

### DISCUSSION THREAD

**Discussion by franz1981** [Hyperfoil/Hyperfoil#275] [Ideas]
Although the original wrk/wrk2 commands doesn't support pipelining OOTB (ie using a cmd parameter), they do support lua scripts that allow configuring pipelining limits easily; AFAIK http 1.x on Hyperfoil already have the pipelining option, hence it's just matter to expose it till wrk/wrk2 as extra-parameter, wdyt?

**Comment by jesperpedersen** [comment]
Moving to a discussion instead.

What do you see as use-cases for having LUA ?

**Reply by franz1981** [reply]
I won't add LUA support, but would expose an extra parameter that doesn't exist on the original wrk/wrk2 command ie `-p <pipelining level>` 
to allow existing (hyperfoil) wrk/wrk2 users to configure pipelining despite LUA support is not present in Hyperfoil (that's probably a too big feature).

**Reply by jesperpedersen** [reply]
Ok, so HTTP 1.1 only with the default of no pipelining.

Feel free to open a `feature` issue ("Support HTTP 1.1 pipelining") and assign yourself.

**Reply by franz1981** [reply]
`Support HTTP 1.1 pipelining` - I see that Hyperfoil already support pipelining on Http 1.1, but is not exposed by `wrk` because the original cmd doesn't have any option to enable it (but the mentioned LUA mechanism, see this [Techempower's LUA script](https://github.com/TechEmpower/FrameworkBenchmarks/blob/a4a52cfe3b4cea997682ed325716f578000d7949/toolset/wrk/pipeline.lua)).

**Reply by jesperpedersen** [reply]
We need a "Discussion" -> "Issue" -> "Pull request" workflow. The PR will reference the Issue in the commit subject.

Full LUA support is a big task - exposing a `-p` option for `wrk` is more doable.
